scripts/Makefile-gaf-reprocess

```diff
@@ -1,9 +1,22 @@
 BASE_NAMES = $(notdir $(wildcard /opt/go-site/annotations/*))
 
-all: $(foreach name,$(BASE_NAMES),/opt/go-site/annotations_new/$(name))
+all: $(foreach name,$(BASE_NAMES),/opt/go-site/annotations_new/$(name)) /opt/go-site/gaferencer-products/all.gaferences.json.gz
 
-/opt/go-site/annotations_new/%.gaf: /opt/go-site/annotations/%.gaf
+/opt/go-site/annotations_new/%.gz: /opt/go-site/annotations_new/%
+	pigz /opt/go-site/annotations_new/$*
+
+/opt/go-site/annotations_new/%.gaf: /opt/go-site/annotations/%.gaf /opt/go-site/gaferencer-products/all.gaferences.json
 	ontobio-parse-assocs.py -f $< -F gaf -o $@ -I /opt/go-site/gaferencer-products/all.gaferences.json --report-md /tmp/report.md --report-json /tmp/report.json convert -to gaf
 
-/opt/go-site/annotations_new/%.gpad: /opt/go-site/annotations/%.gpad
+/opt/go-site/annotations_new/%.gpad: /opt/go-site/annotations/%.gpad /opt/go-site/gaferencer-products/all.gaferences.json
 	ontobio-parse-assocs.py -f $< -F gpad -o $@ -I /opt/go-site/gaferencer-products/all.gaferences.json --report-md /tmp/report.md --report-json /tmp/report.json convert -to gpad
+
+/opt/go-site/annotations/%:
+	unpigz /opt/go-site/annotations/$*.gz
+
+/opt/go-site/gaferencer-products/all.gaferences.json.gz: /opt/go-site/gaferencer-products/all.gaferences.json
+	pigz /opt/go-site/gaferencer-products/all.gaferences.json
+
+/opt/go-site/gaferencer-products/all.gaferences.json:
+	tar -xvf /opt/go-site/gaferencer-products/gaferences.json.tgz -C /opt/go-site/gaferencer-products
```

Comment:
I feel confused about this. This is implying that presumably we downloaded a `all.gaferences.json.tgz` from skyhook, and that now we're appending to it? Why do we have an `all.gaferences.json.tgz` if doesn't already include all the gaferences from the products? Does the mega-make step upload the individual gaferences and bundle them together? Or does the bundling only happen here? In which case we shouldn't have acces to a `gaferences.tgz` yet, right?

Reply:
Referring to https://github.com/geneontology/go-site/issues/1484#issuecomment-639177452:
> preserve the gafrencer products, tarballed and gzipped, from the mega-step step in skyhook (/products/gaferencer/*)

This tarball is `gaferences.json.tgz`, generated and pushed to skyhook in the mega-make step. And then
> pull in gafrencer tarball and convert those files into a single JSON file

This converted single JSON is `gaferences.json.gz` (aka `all.gaferences.json.gz` in the pipeline). This `all.gaferences.json` is what's used by `ontobio-parse-assocs.py`, right?

Reply:
Oh yeah, and `gaferences.json.tgz` is rsync'd in from skyhook (along with `annotations/*`) right before the `make` cmd:
https://github.com/geneontology/pipeline/blob/f8dc05dc48e7fa9c1f7e55eb1f25f5e49a72a5da/Jenkinsfile#L675-L676
So it should be accessible by `make`.

Reply:
Okay just so I have this straight: so when we download, all.gaferences.json.tgz is the bag of all the different gaferencer products from the megam-make step. Then when unzipped, it pops out a bunch files that are the gaferences output from the metagstep, unzipped. Then we concatenate into all.gaferences.json, to be used in the ontobio command. Then we zip all.gaferences.json into all.gaferences.json.tgz, which only includes the one mega gaferences file? 

Is there a reason we shouldn't/couldn't do the concat *in* the mega-make? And then this step just has to download the one gaferences file essentially?

Reply:
@dougli1sqrd Oh sure, I think it can be concatenated in the mega-make step. I'll try it out!

And right about everything else! It does sound pretty convoluted, doesn't it.